src/utils/sequencer/client.go

```diff
@@ -87,3 +100,88 @@ func (self *Client) Upload(ctx context.Context, item *bundlr.BundleItem, mode ty
 
 	return self.UploadReader(ctx, reader, mode)
 }
+
+func (self *Client) UploadInteraction(ctx context.Context, input json.Marshaler, options types.WriteInteractionOptions, signer bundlr.Signer) (out string, err error) {
+	if options.ContractTxId == "" {
+		err = errors.New("Contract tx id not passed")
+		return
+	}
+
+	bundleItem := new(bundlr.BundleItem)
+
+	parsedInput, err := json.Marshal(input)
+	if err != nil {
+		return
+	}
+
+	if len(options.Tags) > 0 {
+		bundleItem.Tags = append(bundleItem.Tags, options.Tags...)
+	}
+
+	bundleItem.Tags = append(bundleItem.Tags,
+		bundlr.Tag{Name: smartweave.TagAppName, Value: smartweave.TagAppNameValue},
+		bundlr.Tag{Name: smartweave.TagAppVersion, Value: smartweave.TagAppVersion},
+		bundlr.Tag{Name: smartweave.TagSDK, Value: smartweave.TagSDKValue},
+		bundlr.Tag{Name: smartweave.TagContractTxId, Value: options.ContractTxId},
+		bundlr.Tag{Name: smartweave.TagInputFormat, Value: smartweave.TagInputFormatTagValue},
+		bundlr.Tag{Name: smartweave.TagInput, Value: string(parsedInput)},
+	)
+	if options.Vrf {
+		bundleItem.Tags = append(bundleItem.Tags,
+			bundlr.Tag{Name: warp.TagRequestVrf, Value: "true"},
+		)
+	}
+
+	var dataFields types.InteractionDataField
+	var data []byte
+
+	tagsBytes, err := bundleItem.Tags.Marshal()
+	if err != nil {
+		return
+	}
+	if len(tagsBytes) > 4096 {
+		inputFormatTagIndex := slices.Index(bundleItem.Tags, bundlr.Tag{Name: smartweave.TagInputFormat, Value: smartweave.TagInputFormatTagValue})
+		bundleItem.Tags = slices.Delete(bundleItem.Tags, inputFormatTagIndex, inputFormatTagIndex+1)
+
+		bundleItem.Tags = append(bundleItem.Tags, bundlr.Tag{Name: smartweave.TagInputFormat, Value: smartweave.TagInputFormatDataValue})
+
+		inputTagIndex := slices.Index(bundleItem.Tags, bundlr.Tag{Name: smartweave.TagInput, Value: string(parsedInput)})
+		bundleItem.Tags = slices.Delete(bundleItem.Tags, inputTagIndex, inputTagIndex+1)
+
+		dataFields.Input = input
+
+	}
+	if options.ManifestData != nil {
+		dataFields.Manifest = options.ManifestData
+	}
+
+	if reflect.DeepEqual(dataFields, types.InteractionDataField{}) {
+		bundleItem.Data = arweave.Base64String(fmt.Sprint(rand.Int())[0:4])
```

Comment:
why rand?

Reply:
Historically each SmartWeave interaction that has input in tags has random 4 bytes of data. In case of data-items I believe it additionaly protects from generating the same txId in case of tags and the rest was the same.